public/components/management/groups/multiple-agent-selector.tsx

```diff
@@ -9,707 +9,897 @@
+      const addedAgents = [];
 
-    modified.forEach(mod => {
-      if (original.filter(e => e.key === mod.key).length === 0) {
-        addedAgents.push(mod);
-      }
-    });
-    original.forEach(orig => {
-      if (modified.filter(e => e.key === orig.key).length === 0) {
-        deletedAgents.push(orig);
-      }
-    });
+      modified.forEach((mod) => {
+        if (original.filter((e) => e.key === mod.key).length === 0) {
+          addedAgents.push(mod);
+        }
+      });
+      original.forEach((orig) => {
+        if (modified.filter((e) => e.key === orig.key).length === 0) {
+          deletedAgents.push(orig);
+        }
+      });
 
-    const addedIds = [...new Set(addedAgents.map(x => x.key))];
-    const deletedIds = [...new Set(deletedAgents.map(x => x.key))];
+      const addedIds = [...new Set(addedAgents.map((x) => x.key))];
+      const deletedIds = [...new Set(deletedAgents.map((x) => x.key))];
 
-    return { addedIds, deletedIds };
-  }
+      return { addedIds, deletedIds };
+    }
+
+    groupBy(collection, property) {
+      try {
+        const values = [];
+        const result = [];
 
-  groupBy(collection, property) {
-    try {
-      const values = [];
-      const result = [];
-
-      for (const item of collection) {
-        const index = values.indexOf(item[property]);
-        if (index > -1) result[index].push(item);
-        else {
-          values.push(item[property]);
-          result.push([item]);
+        for (const item of collection) {
+          const index = values.indexOf(item[property]);
+          if (index > -1) result[index].push(item);
+          else {
+            values.push(item[property]);
+            result.push([item]);
+          }
         }
+        return result.length ? result : false;
+      } catch (error) {
+        return false;
       }
-      return result.length ? result : false;
-    } catch (error) {
-      return false;
     }
-  }
 
-  async saveAddAgents() {
-    const itemsToSave = this.getItemsToSave();
-    const failedIds = [];
-    try {
-      this.setState({ savingChanges: true });
-      if (itemsToSave.addedIds.length) {
-        const addResponse = await WzRequest.apiReq(
-          'PUT',
-          `/agents/group`, {
-          params: {
-            group_id: this.props.currentGroup.name,
-            agents_list: itemsToSave.addedIds.toString()
+    async saveAddAgents() {
+      const itemsToSave = this.getItemsToSave();
+      const failedIds = [];
+      try {
+        this.setState({ savingChanges: true, initState: false });
+        if (itemsToSave.addedIds.length) {
+          const addResponse = await WzRequest.apiReq('PUT', `/agents/group`, {
+            params: {
+              group_id: this.props.currentGroup.name,
+              agents_list: itemsToSave.addedIds.toString(),
+            },
+          });
+
+          if (addResponse.data.data.failed_ids) {
+            failedIds.push(...addResponse.data.data.failed_ids);
```

Comment:
This code is never executed. An error in the assignment of `addResponse` raises an exception that skips this section to the `catch` phase, meaning that `failedIds` is never updated.